tests/roles/dataplane_adoption/tasks/main.yaml

```diff
@@ -279,6 +269,20 @@
             edpm_ovn_ofctrl_wait_before_clear: 8000
     EOF
 
+
+- name: check ovs external-ids with os-diff before deployment
+  no_log: "{{ use_no_log }}"
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ os_diff_dir }}/os-diff diff {{ os_diff_data_dir }}/tripleo/ovs_external_ids/standalone/ovs_external_ids.json edpm-crd.yaml --crd --service ovs_external_ids -f ${PWD}/{{ os_diff_dir }}/config.yaml
```

Comment:
This fails as 

TASK [dataplane_adoption : check ovs external-ids with os-diff before deployment] ***
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "set -euxo pipefail\n\ntmp/os-diff/os-diff diff tmp/os-diff/tripleo/ovs_external_ids/standalone/ovs_external_ids.json edpm-crd.yaml --crd --service ovs_external_ids -f ${PWD}/tmp/os-diff/config.yaml\n", "delta": "0:00:00.005404", "end": "2024-05-14 12:53:42.611228", "msg": "non-zero return code", "rc": 1, "start": "2024-05-14 12:53:42.605824", "stderr": "+ tmp/os-diff/os-diff diff tmp/os-diff/tripleo/ovs_external_ids/standalone/ovs_external_ids.json edpm-crd.yaml --crd --service ovs_external_ids -f /home/ospng/data-plane-adoption/tests/playbooks/tmp/os-diff/config.yaml\nError: unknown flag: --crd\nUsage:\n  os-diff diff [path1] [path2] [flags]\n\nFlags:\n      --debug              Enable debug.\n      --file1-cmd string   Remote command for the file1 configuration file.\n      --file2-cmd string   Remote command for the file2 configuration file.\n  -h, --help               help for diff\n      --quiet              Do not print difference on the console and use logs report, only for files comparison\n      --remote             Run the diff remotely.\n\nGlobal Flags:\n  -c, --config string   Config file (default is $PWD/config.ini) (default \"os-diff.cfg\")", "stderr_lines": ["+ tmp/os-diff/os-diff diff tmp/os-diff/tripleo/ovs_external_ids/standalone/ovs_external_ids.json edpm-crd.yaml --crd --service ovs_external_ids -f /home/ospng/data-plane-adoption/tests/playbooks/tmp/os-diff/config.yaml", "Error: unknown flag: --crd", "Usage:", "  os-diff diff [path1] [path2] [flags]", "", "Flags:", "      --debug              Enable debug.", "      --file1-cmd string   Remote command for the file1 configuration file.", "      --file2-cmd string   Remote command for the file2 configuration file.", "  -h, --help               help for diff", "      --quiet              Do not print difference on the console and use logs report, only for files comparison", "      --remote             Run the diff remotely.", "", "Global Flags:", "  -c, --config string   Config file (default is $PWD/config.ini) (default \"os-diff.cfg\")"], "stdout": "", "stdout_lines": []}


I guess it needs to be reverted

Reply:
What's strange though is that this PR passed CI at merge time. So perhaps we could disable the failing test if we need to, but i wouldn't revert 418 as some of the refactorings (e.g. the order in which things happen) are quite fundamental and i don't think reverting them would be necessary.

Reply:
I don't think we should revert this PR, the issue pointed here is because the version of os-diff used is very old:  9bfd05cafd41d46df3a0c94b5ac781b54646eea6 from Mar 1.
According to:
https://review.rdoproject.org/zuul/builds?job_name=data-plane-adoption-osp-17-to-extracted-crc&skip=0
The jobs are green, and the only red job I'm seing is not related.